src/Compilers/CSharp/Test/Emit3/Semantics/DictionaryExpressionTests.cs

```diff
@@ -4299,6 +4440,395 @@ public V this[K key]
+                        Write("ICollection<T>.CopyTo(..., 0): ");
+                        a = new T[length];
+                        ((ICollection<T>)x).CopyTo(a, 0);
+                        a.Report(includeType: true);
+                        WriteLine();
+                        WriteLine("ICollection<T>.Remove(value): {0}", Invoke(() => ((ICollection<T>)x).Remove(value)));
+                    }
+                    static string Invoke<T>(Func<T> f)
+                    {
+                        try
+                        {
+                            var value = f();
+                            return value is null ? "null" : value.ToString();
+                        }
+                        catch (Exception e)
+                        {
+                            return e.GetType().FullName;
+                        }
+                    }
+                    static string Invoke(Action a)
+                    {
+                        try
+                        {
+                            a();
+                            return "completed";
+                        }
+                        catch (Exception e)
+                        {
+                            return e.GetType().FullName;
+                        }
+                    }
+                }
+                """;
+            var verifier = CompileAndVerify(
+                [source, s_collectionExtensions],
+                expectedOutput: $$"""
+                    System.Collections.ObjectModel.ReadOnlyDictionary<System.Int32, System.String>: IEnumerable.GetEnumerator(): (System.Collections.ObjectModel.ReadOnlyDictionary<System.Int32, System.String>) [], 
+                    ICollection.Count: 0
+                    ICollection.IsSynchronized: False
+                    ICollection.SyncRoot == (object)x: False
+                    ICollection.CopyTo(..., 0): (System.Collections.Generic.KeyValuePair<System.Int32, System.String>[]) [], 
+                    IEnumerable<T>.GetEnumerator(): (System.Collections.ObjectModel.ReadOnlyDictionary<System.Int32, System.String>) [], 
+                    IReadOnlyCollection<T>.Count: 0
+                    ICollection<T>.Count: 0
+                    ICollection<T>.IsReadOnly: True
+                    ICollection<T>.Add(value): System.NotSupportedException
+                    ICollection<T>.Clear(): System.NotSupportedException
+                    ICollection<T>.Contains(value): False
+                    ICollection<T>.CopyTo(..., 0): (System.Collections.Generic.KeyValuePair<System.Int32, System.String>[]) [], 
+                    ICollection<T>.Remove(value): System.NotSupportedException
+                    IDictionary.IsFixedSize: True
+                    IDictionary.IsReadOnly: True
+                    IDictionary.this[k]: {{(ExecutionConditionUtil.IsMonoOrCoreClr ? "null" : "System.Collections.Generic.KeyNotFoundException")}}
```

Comment:
Note, the implementation of `IDictionary.this[TKey key]` is different between .NET Framework and .NET Core. On .NET Core, the indexer returns `null` rather than throwing `KeyNotFoundException` if the key is missing.

Reply:
Weird! Are you aware of any documentation about this?

Reply:
> Are you aware of any documentation about this?

No, I have not seen any mention of this. It is not mentioned in the current [learn.microsoft.com](https://learn.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlydictionary-2.system-collections-idictionary-item?view=net-9.0) documentation for instance.